### PR TITLE
Prevent browser autofill from appearing on an element with autocomplete directive

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
#93 

Chrome ignores `autocomplete="off"`. This is a known chrome "bug": https://bugs.chromium.org/p/chromium/issues/detail?id=914451

To make things more challenging, other browsers interpret these rules differently. For example, Firefox respects `off` while Safari does not.

More info on the `autocomplete` attribute:
https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#The_autocomplete_attribute_and_login_fields

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix)

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 1]
```diff
@@ -214,7 +216,12 @@ export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, Control
   public onValidatorChange = () => {};
 
   private setAttributes(element: any): void {
-    this.renderer.setAttribute(element, 'autocomplete', 'off');
+    /**
+     * Need a unique value for autocomplete to be disabled on Chrome.
+     * https://bugs.chromium.org/p/chromium/issues/detail?id=914451
+     */
+    const id = ++uniqueId;
+    this.renderer.setAttribute(element, 'autocomplete', `skyux-autocomplete-${id}-off`);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:224]: If this works for Chrome, does it work for all the other browsers, too? I wouldn't want to introduce a hack for one browser at the cost of the others...

Comment (jeffbdye) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:224]: @Blackbaud-SteveBrush we've employed a similar approach to disable autocomplete that works across browsers, I'll send you the link to the internal repo.

**Comment by Blackbaud-AlexKingman** [comment]
Just found some more complications with Firefox. Closing this PR until we find a more compatible cross-browser solution.

**Comment by Blackbaud-AlexKingman** [comment]
`autocomplete` is now set to `new-password`. While not semantic, we felt this was the best move forward to deal with each browser interpreting the rules for `autocomplete` differently.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix)

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 2]
```diff
@@ -214,7 +214,15 @@ export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, Control
...
   private setAttributes(element: any): void {
-    this.renderer.setAttribute(element, 'autocomplete', 'off');
+    /**
+     * Modern browsers interpret autocomplete rules differently
+     * and sometimes completely ignore the 'off' value. As a workaround,
+     * 'new-password' will prevent autocomplete in FF 76.0.1, Chrome 81.0, and Edge 81.0.
+     * For preventing autocomplete in Safari, apply directive to a textarea element.
+     * https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion
+     */
+    this.renderer.setAttribute(element, 'autocomplete', 'new-password');
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:224]: Thanks for the comments! Would it be too much trouble to add a unit test to confirm the attribute is set correctly, as an added protection from a future engineer removing it?

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:224]: Good point. Done.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-contry-field-autocomplete-fix)

**Pull request merged**